src/drivers/intel/ssp/ssp.c

```diff
@@ -766,6 +728,9 @@ static void ssp_start(struct dai *dai, int direction)
 
 	spin_lock(&dai->lock);
 
+	/* request mclk/bclk */
+	ssp_pre_start(dai);
```

Comment:
You should have a first patch that moves all the code from ssp_set_config() to ssp_pre_start(), and then add the M/N changes.

Reply:
do we want to do this for pause/relkease as well?

Reply:
> You should have a first patch that moves all the code from ssp_set_config() to ssp_pre_start(), and then add the M/N changes.

Do you mean split this commit to 2? I thought about that, but compiler may complaint for the first commit that "function defined but not used".

Reply:
> do we want to do this for pause/release as well?

It will be called when ssp_start/stop is called with this change applied, so it will not be applied at ssp_pause() at the moment. For this pause/release scenario, it could be tricky as we don't release clear the RXEN bit at the moment, so forcing to release the bclk might lead to jitter (samples lost), but we are good with this series as we don't do this for pause.